Logging/ARKLogMessage.h

```diff
@@ -27,15 +27,23 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface ARKLogMessage : NSObject <NSCopying, NSSecureCoding>
 
+/// @warning Do not use this initializer.
+/// @raises NSInternalInconsistencyException This is not a valid initializer.
+- (instancetype)init;
```

Comment:
I agree this is a better design, but this is a pretty substantial refactor. I'm just trying to make a targeted incremental change to make this a bit better. Could we have the protocol approach be our plan of record going forwards?